ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -99,63 +127,39 @@ export default class PermissionConnect extends Component {
     }, () => this.props.history.push(this.props.confirmPermissionPath))
   }
 
-  redirectFlow (accepted) {
-    const { history, location, confirmPermissionPath } = this.props
+  redirect (approved) {
+    const { history, hasPendingPermissionsRequests } = this.props
 
     this.setState({
       redirecting: true,
-      permissionAccepted: accepted,
+      permissionsApproved: approved,
     })
     this.removeBeforeUnload()
 
-    if (getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
-      setTimeout(async () => {
+    const doRedirect = () => {
```

Comment:
Nit: Better to move this function to module scope or make it a static method, so it's not constructed anew each time this is called.

Reply:
d8e7fb4